### ISSUE TITLE
add sample for handling android-23 permission flow thru native-activity

### ISSUE THREAD

**Issue by proppy** [googlesamples/android-ndk#114]
#96 show how to handle them from java, it would be nice to have a sample that show how to initial a permission flow from C++.

/cc @Whiteseeker

**Comment by proppy** [comment]
This need to show how to extend native activity to implement the `onRequestPermissionsResult` callback.

**Comment by claywilkinson** [comment]
In the Unity plugin, I wrote a [Fragment](https://github.com/playgameservices/play-games-plugin-for-unity/blob/master/source/SupportLib/PlayGamesPluginSupport/src/main/java/com/google/games/bridge/TokenFragment.java) that prompts for the permission if needed to access the auth token.  It works, I think it could be the model for a more general purpose approach for Native apps to work with the new permission model

**Comment by ggfan** [comment]
run-time permission for audio are merged; that's is all needed for the repo now.  Like to close this one, any thoughts again it?  thanks

**Comment by proppy** [comment]
@ggfan, I think @claywilkinson approach is useful for app w/ nativeactivity when you can override dynamically the methods of the activity class to add the permission callback. But you can add dynamically a fragment instead.

**Comment by ggfan** [comment]
correct, the callback mechanism is useful for the native-activity derived apps.  Theoetically run-time permission in that category could be done with:
1) derive a class from NativeActivity, and request the permission at the startup
2) use callback from jni side to take a native-callback pointer and do as it is processed in claywilkinson's  implementation.

we might cross this one later.

**Comment by Cwiiis** [comment]
Just ran into this one and there hasn't been any activity here for a while... Did I just manage to miss the example, or is this still an issue?

**Comment by ggfan** [comment]
@Cwiis The example of enabling it from application side could be found in camera sample:
 https://github.com/googlesamples/android-ndk/blob/master/camera/basic/src/main/java/com/sample/camera/basic/CameraActivity.java    
If you are expecting to have NativeActivity to have an interface to request permission automatically, no it is not there yet as I understand.

**Comment by ggfan** [comment]
I like to close this one and if you have strong desire to add something in, yeah add a PR is great. I do look forward to having NativeActivity to have this functionality built-in: so NativeActivity keeps its cool of 0 java code in side the application.

**Comment by Cwiiis** [comment]
This ought to be reopened - this bug is specifically about a NativeActivity example, not a Java example. Even if it involves some small amount of Java, an example of how to do this from a NativeActivity-based application would still be very useful.

**Comment by ggfan** [comment]
the camera sample is derived from NativeActivity with camera permission request.
endless-tunnel, I believe, also needs disk write permission ( but never add it yet ) and should be added. Once that is done, enough usage of it inside sample.  But if you are looking for the integration inside NativeActivity for permission, that is not in the sample ( I would expect that would goes into NativeActivity's internal java code ).  Let me know I understood this wrong.  thx

**Comment by Cwiiis** [comment]
I think I misunderstood the camera sample, and it's enough for now - but ideally, you wouldn't have to extend NativeActivity to process permissions, I think it ought to handle that case and provide a native API for it. I suppose this is a problem with the Android ndk though, and not these samples. Thanks for your patience!

**Comment by sadernalwis** [comment]
this should work to bring permission flow to c++ until a proper sample comes up.
https://github.com/googlesamples/android-ndk/issues/636
unfortunately i need some help to make it work. this should work, theoretically.

**Comment by sadernalwis** [comment]
this issue should be re-opened. NDK cannot do anything substantial without permissions. going back and forth to & from Java just for permission when an app architecture relies on turning runtime-permission  on and off will throttle the performance advantage expected from using NDK!/?
